pkg/controller/device.go

```diff
@@ -45,6 +46,143 @@ func (cloud *CloudCtx) StateUpdate(dev *device.Ctx) (err error) {
 	return fmt.Errorf("cannot load config %s", edenConfig)
 }
 
+type idCheckable interface {
+	GetId() *config.UUIDandVersion
+}
+type uuidCheckable interface {
+	GetUuidandversion() *config.UUIDandVersion
+}
+
+func getID(uuidAndVersion idCheckable) (uuid.UUID, error) {
+	if uuidAndVersion == nil {
+		return uuid.Nil, fmt.Errorf("nil object")
+	}
+	if uuidAndVersion.GetId() == nil {
+		return uuid.Nil, fmt.Errorf("nil UUIDandVersion")
+	}
+	if uuidAndVersion.GetId().GetUuid() == "" {
+		return uuid.Nil, fmt.Errorf("nil UUIDandVersion")
+	}
+	return uuid.FromString(uuidAndVersion.GetId().GetUuid())
+}
+
+func getUUID(uuidAndVersion uuidCheckable) (uuid.UUID, error) {
+	if uuidAndVersion == nil {
+		return uuid.Nil, fmt.Errorf("nil object")
+	}
+	if uuidAndVersion.GetUuidandversion() == nil {
+		return uuid.Nil, fmt.Errorf("nil UUIDandVersion")
+	}
+	if uuidAndVersion.GetUuidandversion().GetUuid() == "" {
+		return uuid.Nil, fmt.Errorf("nil UUIDandVersion")
+	}
+	return uuid.FromString(uuidAndVersion.GetUuidandversion().GetUuid())
+}
+
+//ConfigParse load config into cloud
+func (cloud *CloudCtx) ConfigParse(config *config.EdgeDevConfig) (device *device.Ctx, err error) {
```

Comment:
small nit: would be useful to be super explicit that this function not so much parses config (there's nothing to parse -- it is all in go structures in config.EdgeDevConfig) but rather populates or builds your internal representation of config from protobuf defined go structure.

So at least lets reflect that in the function's name. On top of that, is there any reason not to use the actual config.EdgeDevConfig as the basis of device.Ctx? Along the lines of:
```
struct device.Ctx {
    config.EdgeDevConfig
}
```

?